src/RelationParser.h

```diff
@@ -0,0 +1,74 @@
+//
+//  RelationParser.h
+//  snowcrash
+//
+//  Created by Pavan Kumar Sunkara on 04/03/15.
+//  Copyright (c) 2015 Apiary Inc. All rights reserved.
+//
+
+#ifndef SNOWCRASH_RELATIONPARSER_H
+#define SNOWCRASH_RELATIONPARSER_H
+
+#include "SectionParser.h"
+#include "RegexMatch.h"
+
+namespace snowcrash {
+
+    /** Link Relation matching regex */
+    const char* const RelationRegex = "^[[:blank:]]*[Rr]elation[[:blank:]]*:" SYMBOL_IDENTIFIER "?$";
+
+    /**
+     *  Relation Section Processor
+     */
+    template<>
+    struct SectionProcessor<Relation> : public SectionProcessorBase<Relation> {
```

Comment:
I start some refactoring there, 
https://github.com/apiaryio/snowcrash/compare/master...klokane:milestone-keywords

hope there will be time to finish :)  

At least we should introduce type traits to `SectionProcessor`